### PR TITLE
feat: add async has,get,set methods

### PR DESCRIPTION
**Pull request by sveisvei**
## JIRA Issue
[COREWEB-39](https://jira.finn.no/browse/COREWEB-39)

## Status
**READY**

## Description
Adds async methods to support setting and gettings files without streams, needed for simpler meta-data storage interface.

## Todos
- [x] Tests
- [x] Documentation

## Related PRs
* https://github.com/asset-pipe/asset-pipe-sink-gcs/pull/46

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,6 +71,19 @@ This module have the following API:
...
 Constructor takes no arguments.
 
+
+### get(key)
+
+Async method for reading a file from memory
+
+### set(key, value)
+
+Async method for writing to file in memory
```
Comment (digitalsadhu) [inline README.md]: s/to file in/a file to

File: README.md [hunk 2]
```diff
@@ -71,6 +71,19 @@ This module have the following API:
...
+
+### has(key)
+
+Async method for checking if file exist
```
Comment (digitalsadhu) [inline README.md]: s/file exist/a file exists

File: lib/sink.js [hunk 3]
```diff
@@ -66,6 +66,22 @@ module.exports = class SinkMem extends EventEmitter {
         this.db = {};
     }
 
+    async get(filename) {
+        return this.db[filename] ? this.db[filename] : null;
```
Comment (digitalsadhu) [inline lib/sink.js]: is null the same response from the gcs sink?

Comment (sveisvei) [reply lib/sink.js]: I guess we should reject....?

**Comment by sveisvei** [comment]
changed .get() to reject if no value.

File: lib/sink.js [hunk 4]
```diff
@@ -66,6 +66,24 @@ module.exports = class SinkMem extends EventEmitter {
         this.db = {};
     }
 
+    async get(fileName) {
+        if (this.db[fileName]) {
+            return this.db[fileName];
+        }
+        throw new Error(`No file with ${fileName}`);
```
Comment (trygve-lie) [inline lib/sink.js]: Typo; "No file with name ...."

Comment (sveisvei) [reply lib/sink.js]: https://github.com/asset-pipe/asset-pipe-sink-mem/commit/1299fe6de2b6d26f9f4121088fac1f7e60531756

**Pull request merged**